backend/app/controllers/uoas.js

```diff
@@ -77,48 +74,119 @@ module.exports = {
 
     insert: function (req, res, next) {
         var thunkQuery = req.thunkQuery;
+        var uoas;
+        var sqlString;
 
         // Verify that the body contains the subjects
         if (req.body.subjects) {
             //convert the input to an array because we want to be able to pass in just one subject
             if (!Array.isArray(req.body.subjects)) {
                 req.body.subjects = [{name: req.body.subjects}];
             }
-            var uoas = req.body.subjects.map((subject) => subject.name);
-            var sqlString = "'" + uoas.toString().replace(/'/g, "''").replace(/,/g, "','") + "'";
+
+            // Check that no blank subject name was passed in
+            for (let i = 0; i < req.body.subjects.length; i++) {
+                if (req.body.subjects[i].name == '') {
+                    throw new HttpError(400, 'Subject Cannot be empty');
```

Comment:
Ideally, if there is an empty string in the list, I say we throw an error and not add any of the other items unless the entire list is formatted correctly.